api/dbadapters/bigquery.ts

```diff
@@ -1,25 +1,29 @@
-import * as protos from "@dataform/protos";
+import { dataform } from "@dataform/protos";
 import * as PromisePool from "promise-pool-executor";
-import { DbAdapter, OnCancel } from "./index";
+import { DbAdapter, OnCancel } from "df/api/dbadapters/index";
+import * as util from "df/api/utils";
 
 const BigQuery = require("@google-cloud/bigquery");
 
 export class BigQueryDbAdapter implements DbAdapter {
-  private profile: protos.IProfile;
+  private bigquery: dataform.IBigQuery;
   private client: any;
   private pool: PromisePool.PromisePoolExecutor;
 
-  constructor(profile: protos.IProfile) {
-    this.profile = profile;
+  constructor(credentials: util.Credentials) {
+    if (!(credentials instanceof dataform.BigQuery)) {
+      throw new Error("Invalid credentials; did not receive a BigQuery protobuf instance.");
+    }
+    this.bigquery = credentials;
     this.client = BigQuery({
-      projectId: profile.bigquery.projectId,
-      credentials: JSON.parse(profile.bigquery.credentials),
+      projectId: this.bigquery.projectId,
+      credentials: JSON.parse(this.bigquery.credentials),
       scopes: ["https://www.googleapis.com/auth/drive"]
     });
     // Bigquery allows 50 concurrent queries, and a rate limit of 100/user/second by default.
     // These limits should be safely low enough for most projects.
     this.pool = new PromisePool.PromisePoolExecutor({
-      concurrencyLimit: this.profile.threads || 16,
```

Comment:
We actually have something on our sprint to actually use this value, can we keep it?

Reply:
Discussed last night but for posterity: I'm removing this because it was part of the Profile, and since it's unused it's easier to remove it for now. We should probably add it back as part of the RunConfig.